[datadog_tag_pipeline_rulesets] add the functionality to allow users decide tag pipeline reorder behaviors (#3320)
DataDog/terraform-provider-datadog: docs/resources/tag_pipeline_rulesets.md

        value = "backend"
      }
    }
  }
}

resource "datadog_tag_pipeline_ruleset" "third" {
  name    = "Enrich Service Metadata"
  enabled = true

  rules {
    name    = "lookup-service"
    enabled = true

    reference_table {
      table_name         = "service_catalog"
      case_insensitivity = true
      if_not_exists      = true
      source_keys        = ["service"]

      field_pairs {
        input_column = "owner_team"
        output_key   = "owner"
      }
    }
  }
}

# Manage the order of tag pipeline rulesets
# Rulesets are executed in the order specified in ruleset_ids
# UI-defined rulesets not in this list will be preserved at the end
resource "datadog_tag_pipeline_rulesets" "order" {
  ruleset_ids = [
    datadog_tag_pipeline_ruleset.first.id,
    datadog_tag_pipeline_ruleset.second.id,
    datadog_tag_pipeline_ruleset.third.id
  ]
}

# ============================================================================
# Example 2: Override UI-defined rulesets (override_ui_defined_resources = true)
# ============================================================================
# When set to true, any rulesets created via the UI that are not defined in Terraform

> # will be automatically deleted during terraform apply.

Reviewer: ```suggestion
# will be automatically deleted during `terraform apply`.
```

- `ruleset_ids` (List of String) The list of Tag Pipeline Ruleset IDs, in order. Rulesets are executed in the order specified in this list.

### Optional

> - `override_ui_defined_resources` (Boolean) Whether to override UI-defined rulesets. When set to true, any rulesets created via the UI that are not defined in Terraform will be deleted and perform reorder based on the rulesets from the terraform. Default is false

Reviewer 2: ```suggestion
- `override_ui_defined_resources` (Boolean) Whether to override rulesets defined in the UI. When set to `true`, any UI-created rulesets not defined in Terraform will be deleted, and the remaining rulesets will be reordered based on the Terraform configuration. Default is `false`.
```
